File: 0024-swap-nodes-in-pairs/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
            // my code learn from swap in k group question
        if(head == null) return null;

        int cnt= 0;
        ListNode curr= head, prev = null;
        while(curr != null && cnt<2){
            cnt++;
            ListNode nextNode = curr.next;
            curr.next = prev;
            prev = curr;
            curr = nextNode;
        }
        head.next = swapPairs(curr);
        return prev;
    }
}

File: 0024-swap-nodes-in-pairs/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* dummyNode = new ListNode(-1);
        dummyNode->next = head;
        ListNode* prev = dummyNode;
        ListNode* curr = head;

        while(curr != NULL && curr->next != NULL){
            ListNode* fNode = curr;
            ListNode* sNode = curr->next;

            prev->next = sNode;
            curr->next = sNode->next;
            sNode->next = fNode;

            prev = fNode;
            curr = fNode->next;
        }
        return dummyNode->next;
    }
};
